src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateClone = Object.assign({}, state);
+  const statesOfSteps = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateClone, action.extraData);
+        statesOfSteps.push({ ...stateClone });
+        continue;
```

Comment:
```suggestion
        Object.assign(stateClone, action.extraData);
        break;
```

you can push once at the end